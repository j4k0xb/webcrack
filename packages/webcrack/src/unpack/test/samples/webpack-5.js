(() => {
  var modules = {
    2: (module, exports, require) => {
      'use strict';
      const lib = require(3);
      console.log(lib);
    },
    3: (module, exports) => {
      'use strict';
      require.r(exports);

      const a = require(4);
      const obj = {
        version: '2.0.0',
      };
      require.d(exports, {
        default: () => a.foo,
        version: () => obj.version,
      });
    },
    4: (module, exports) => {
      'use strict';
      require.r(exports);

      const b = 2;
      require.d(exports, {
        foo: () => b,
        obj: () => x,
      });

      var x = {};
      require.r(x);
      require.d(x, {
        Console: () => bar,
      });
      function bar() {}
    }
  };
  var installedModules = {};
  function __webpack_require__(moduleId) {
    var _0x1d30a7 = installedModules[moduleId];
    if (_0x1d30a7 !== undefined) {
      return _0x1d30a7.exports;
    }
    var module = (installedModules[moduleId] = {
      id: moduleId,
      loaded: false,
      exports: {},
    });
    modules[moduleId].call(
      module.exports,
      module,
      module.exports,
      __webpack_require__
    );
    module.loaded = true;
    return module.exports;
  }
  __webpack_require__.c = installedModules;
  __webpack_require__.n = module => {
    var getter =
      module && module.__esModule ? () => module.default : () => module;
    __webpack_require__.d(getter, {
      a: getter,
    });
    return getter;
  };
  __webpack_require__.d = (exports, definition) => {
    for (var _0x43c485 in definition) {
      if (
        __webpack_require__.o(definition, _0x43c485) &&
        !__webpack_require__.o(exports, _0x43c485)
      ) {
        Object.defineProperty(exports, _0x43c485, {
          enumerable: true,
          get: definition[_0x43c485],
        });
      }
    }
  };
  __webpack_require__.o = (object, property) =>
    Object.prototype.hasOwnProperty.call(object, property);
  __webpack_require__.r = exports => {
    if (typeof Symbol != 'undefined' && Symbol.toStringTag) {
      Object.defineProperty(exports, Symbol.toStringTag, {
        value: 'Module',
      });
    }
    Object.defineProperty(exports, '__esModule', {
      value: true,
    });
  };
  if (__webpack_require__ !== undefined) {
    __webpack_require__.ab = __dirname + '/native_modules/';
  }
  var entryModule = __webpack_require__((__webpack_require__.s = 2));
  module.exports = entryModule;
})();
