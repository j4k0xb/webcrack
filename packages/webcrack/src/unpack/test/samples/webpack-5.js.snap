WebpackBundle {
  "entryId": "2",
  "modules": Map {
    "2" => WebpackModule {
      "ast": const lib = require("./3.js");
console.log(lib);,
      "id": "2",
      "isEntry": true,
      "path": "./index.js",
    },
    "3" => WebpackModule {
      "ast": import * as a from "./4.js";
const obj = {
  version: "2.0.0"
};
export let version = obj.version;
export default a.foo;,
      "id": "3",
      "isEntry": false,
      "path": "./3.js",
    },
    "4" => WebpackModule {
      "ast": export const foo = 2;
export var obj = {
  Console: bar
};
function bar() {},
      "id": "4",
      "isEntry": false,
      "path": "./4.js",
    },
  },
  "type": "webpack",
}